test/TestHelper.jl

```diff
@@ -118,13 +119,41 @@ function gen_ncdata_time(FT, path, varname, val)
     # Define variables
     lon = NCDatasets.defVar(nc, "lon", FT, ("lon",))
     lat = NCDatasets.defVar(nc, "lat", FT, ("lat",))
-    time = NCDatasets.defVar(nc, "time", Float64, ("time",))
+    time = NCDatasets.defVar(nc, "time", Dates.DateTime.([2020.0, 2021.0]), ("time",))
```

Comment:
why was this change needed?

Reply:
it looks like `gen_ncdata_time` and `gen_ncdata_date` were only used in the regridder tests so now they're unused and can be removed

Reply:
That change was made because the `read_available_dates` in ClimaUtilities incorrectly interprets the time dimension when it is a Number. The unused `gen_ncdata_time` and `gen_ncdata_date` are now removed.